pkg/controller/operators/catalog/operator.go

```diff
@@ -622,6 +628,35 @@ func (o *Operator) syncConnection(logger *logrus.Entry, in *v1alpha1.CatalogSour
 	return
 }
 
+func (o *Operator) waitForBackingPod(logger *logrus.Entry, in *v1alpha1.CatalogSource) (out *v1alpha1.CatalogSource, continueSync bool, syncError error) {
```

Comment:
It could be, but I'm not sure it should be. With the way the code is structured and higher level operations (which I consider pod checking to be) present in the main operator code, it seems best to me where it is. There is a comment to add memoization for ReconcileForSource, which if done later on would make the requested refactoring a bit less invasive.